pkg/guacanalytics/helpers.go

```diff
@@ -0,0 +1,317 @@
+
+		if err != nil {
+			return fmt.Errorf("error in ingesting EqualArtifact for HashEqual: %v\n", err)
+		}
+
+		_, err = model.HashEqual(ctx, client, *ingest.Artifact, *ingest.EqualArtifact, *ingest.HashEqual)
+
+		if err != nil {
+			return fmt.Errorf("error in ingesting HashEqual: %v\n", err)
+		}
+	}
+	return nil
+}
+
+func ingestPointOfContact(ctx context.Context, client graphql.Client, graph assembler.IngestPredicates) error {
+	for _, ingest := range graph.PointOfContact {
+		var err error
+
+		if ingest.Src != nil {
+			_, err = model.IngestSource(ctx, client, *ingest.Src)
+		} else if ingest.Pkg != nil {
+			_, err = model.IngestPackage(ctx, client, *ingest.Pkg)
+
+		} else {
+			_, err = model.IngestArtifact(ctx, client, *ingest.Artifact)
+		}
+
+		if err != nil {
+			return fmt.Errorf("error in ingesting pkg/src/artifact PointOfContact: %v\n", err)
+		}
+
+		if ingest.Src != nil {
+			_, err = model.PointOfContactSrc(ctx, client, *ingest.Src, *ingest.PointOfContact)
+		} else if ingest.Pkg != nil {
+			_, err = model.PointOfContactPkg(ctx, client, *ingest.Pkg, ingest.PkgMatchFlag, *ingest.PointOfContact)
+		} else {
+			_, err = model.PointOfContactArtifact(ctx, client, *ingest.Artifact, *ingest.PointOfContact)
+		}
+
+		if err != nil {
+			return fmt.Errorf("error in ingesting PointOfContact: %v\n", err)
+		}
+	}
+	return nil
+}
+
+func IngestTestData(ctx context.Context, client graphql.Client, graph assembler.IngestPredicates) error {
+	if len(graph.IsDependency) > 0 {
+		err := ingestIsDependency(ctx, client, graph)
+		if err != nil {
+			return err
+		}
+	}
+
+	if len(graph.IsOccurrence) > 0 {
+		err := ingestIsOccurrence(ctx, client, graph)
+		if err != nil {
+			return err
+		}
+	}
+
+	if len(graph.HasSlsa) > 0 {
+		err := ingestHasSLSA(ctx, client, graph)
+		if err != nil {
+			return err
+		}
+	}
+
+	if len(graph.CertifyGood) > 0 {
+		err := ingestCertifyGood(ctx, client, graph)
+		if err != nil {
+			return err
+		}
+	}
+
+	if len(graph.HasSourceAt) > 0 {
+		err := ingestHasSourceAt(ctx, client, graph)
+		if err != nil {
+			return err
+		}
+	}
+
+	if len(graph.PkgEqual) > 0 {
+		err := ingestPkgEqual(ctx, client, graph)
+		if err != nil {
+			return err
+		}
+	}
+
+	if len(graph.HashEqual) > 0 {
+		err := ingestHashEqual(ctx, client, graph)
+		if err != nil {
+			return err
+		}
+	}
+
+	if len(graph.PointOfContact) > 0 {
+		err := ingestPointOfContact(ctx, client, graph)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// This function return matching packageName and/or packageVersion node IDs depending on if you specified to only find name nodes or version nodes
+func GetPackageIDs(ctx context.Context, gqlClient graphql.Client, nodeType *string, nodeNamespace string, nodeName string, nodeVersion *string, justFindVersion bool, justFindName bool) ([]*string, error) {
+	var pkgFilter model.PkgSpec
+	if nodeVersion != nil {
+		pkgFilter = model.PkgSpec{
```

Comment:
subpath and qualifiers?

Reply:
same as here: https://github.com/guacsec/guac/pull/1129#discussion_r1286368839

Reply:
For this case, the GetPackageIDs is only used for the purposes in tests and right now the tests don't use subpath and qualifiers but I added a TODO.